src/enableDisableRules.ts

```diff
@@ -63,10 +72,31 @@ export class EnableDisableRulesWalker extends SkippableTokenAwareRuleWalker {
                     if (!(ruleToAdd in this.enableDisableRuleMap)) {
                         this.enableDisableRuleMap[ruleToAdd] = [];
                     }
-                    this.enableDisableRuleMap[ruleToAdd].push({
-                        isEnabled: isEnabled,
-                        position: startingPosition,
-                    });
+                    if (isCurrentLine) {
+                        // start at the beginning of the current line
+                        this.enableDisableRuleMap[ruleToAdd].push({
+                            isEnabled: isEnabled,
+                            position: this.getStartOfLinePosition(node, startingPosition),
+                        });
+                        // end at the beginning of the next line
+                        this.enableDisableRuleMap[ruleToAdd].push({
+                            isEnabled: !isEnabled,
+                            position: scanner.getTextPos() + 1,
```

Comment:
Oh sorry, i see what you were trying to say now. Either version should yield the same position. `scanner.getTextPos()` returns the position of the end of the text, the `+1` is just to bump the position to the beginning of the next line. I guess that the current expression may be a little terse, I could add an extra comment for clarity if you like?